Sources/unitfrmpgndatabase.pas

```diff
@@ -0,0 +1,156 @@
+
+  end;
+
+var
+  frmPGNDatabase: TfrmPGNDatabase;
+  My_Grid_Highlight_BackGround_Color:Tcolor=ClGreen;
+  My_Grid_Highlight_Font_Color:Tcolor=clYellow;
+  My_Grid_Highlight_Font_Multiplier:Real=1.5;  // FOnt Multiplier for selected row
+  LastRow : Integer;
+implementation
+
+uses
+  MainUnit, GraphUtil, LCLIntf;
+{$R *.lfm}
+
+{ TfrmPGNDatabase }
+
+procedure TfrmPGNDatabase.gridPGNDatabaseDblClick(Sender: TObject);
+begin
+    mniLoadGameClick(nil);
+end;
+procedure TfrmPGNDatabase.gridPGNDatabaseDrawCell(Sender: TObject; aCol,
+  aRow: Integer; aRect: TRect; aState: TGridDrawState);
+begin
+  If Sender is TStringGrid then
+  begin
+    With (sender as TStringGrid) Do
+    Begin
+      Canvas.Brush.Style := bsSolid;
+      canvas.Font.Color:=clWhite;
+      if arow=row then       // Selected Row
+      begin
+        FocusColor:=My_Grid_Highlight_BackGround_Color;
+        Canvas.Brush.Color:=My_Grid_Highlight_BackGround_Color;
+        canvas.Font.Color:=My_Grid_Highlight_Font_Color;
+        canvas.Font.Size:=round(font.Size*My_Grid_Highlight_Font_Multiplier);
+      end
+      else
+      begin
+        canvas.Font.Color:=font.Color;
+        Canvas.Brush.Color:=color;
+        canvas.Font.Size:=font.Size;
+      end;
+      defaultdrawcell(Acol, Arow, arect, astate);
+    end;
+  end;
+end;
+
+procedure TfrmPGNDatabase.gridPGNDatabaseMouseMove(Sender: TObject;
+  Shift: TShiftState; X, Y: Integer);
+var
+  R, C:        Integer;
+  Start, Size: Integer;
+  i:           Integer;
+begin
+  gridPGNDatabase.MouseToCell(X, Y, c, r);
+  LastRow := r;
+end;
+
+procedure TfrmPGNDatabase.gridPGNDatabasePrepareCanvas(Sender: TObject; aCol,
+  aRow: Integer; aState: TGridDrawState);
+var
+  P: TPoint;
+  c, r: LongInt;
+  obe: Boolean;
+begin
+  P := gridPGNDatabase.ScreenToClient(Mouse.CursorPos);
+  if (aState = []) or (gdFixed in aState) then
+  begin
+    obe := gridPGNDatabase.AllowOutBoundEvents;
+    gridPGNDatabase.AllowOutBoundEvents := false;
+    gridPGNDatabase.MouseToCell(P.X, P.Y, c, r);
+    gridPGNDatabase.AllowOutBoundEvents := obe;
+    if (aRow = r) and (aRow >= gridPGNDatabase.FixedRows) then
+    begin
+      if aState = [] then
+        gridPGNDatabase.Canvas.Brush.Color := rgb(232, 232, 232)
+      else
+        gridPGNDatabase.Canvas.Brush.Color := GetShadowColor(ColorToRGB(gridPGNDatabase.FixedColor), -20);
+    end;
+  end;
+  gridPGNDatabase.Invalidate();
+  gridPGNDatabase.Refresh();
+end;
+
+procedure TfrmPGNDatabase.mniLoadGameClick(Sender: TObject);
+begin
+   BoardForm.Caption := gridPGNDatabase.Rows[gridPGNDatabase.Row].Strings[1];
+   if (BoardForm.Caption = '') then
+   begin
+     BoardForm.Caption := 'Board';
+   end;
+   MainForm.FGame.ChessNotation.PGNString := gridPGNDatabase.Rows[gridPGNDatabase.Row].Strings[0] + ' ' + gridPGNDatabase.Rows[gridPGNDatabase.Row].Strings[3];
+   while (not MainForm.FGame.ChessNotation.Iterator.IsFirst) do
+   begin
+      MainForm.FGame.ChessNotation.Iterator.PrevMove();
```

Comment:
Is all this switching step-by-step required?

For Paste PGN code, it was not:
https://github.com/alex65536/Chess256/blob/master/Sources/NotationForms.pas#L325-L333 

Also, notation has
https://github.com/alex65536/Chess256/blob/master/Sources/ChessNotation.pas#L146

Reply:
Possibly, it's better to leave the cursor at the end of the game when you load it.

Reply:
I didn't want the user to see the end of the game and spoil it so to speak.